# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1331591774 14400
#      Mon Mar 12 18:36:14 2012 -0400
# Node ID 008a29999c3d39490b80fd17dc45ae1a2fc6cf02
# Parent  05635e4c605a601f10b932f465c8f92e42e633de
Save the papersize property in the proper orientation.

* graphics.cc (figure::properties::update_papersize)

diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -3651,16 +3651,21 @@ figure::properties::update_papersize (vo
       else if (std::abs (sz(0) - 22.0) + std::abs (sz(1) - 34.0) < tol)
         typ = "d";
       else if (std::abs (sz(0) - 34.0) + std::abs (sz(1) - 43.0) < tol)
         typ = "e";
       // Call papertype.set rather than set_papertype to avoid loops between
       // update_papersize and update_papertype
       papertype.set (typ);
     }
+  if (get_paperorientation () == "landscape")
+    {
+      std::swap (sz(0), sz(1));
+      papersize.set (octave_value (sz));
+    }
 }
 
 void
 figure::properties::update_paperorientation (void)
 {
   std::string porient = get_paperorientation ();
   Matrix sz = get_papersize ().matrix_value ();
   Matrix pos = get_paperposition ().matrix_value ();
